resolwe_bio/processes/chip_seq/chipqc.py

```diff
@@ -0,0 +1,201 @@
+"""Calculate quality metrics for ChIPseq data.."""
+
+from pathlib import Path
+from shutil import copyfile
+
+from plumbum import TEE
+
+from resolwe.process import (
+    Process,
+    SchedulingClass,
+    Persistence,
+    Cmd,
+
+    StringField,
+    BooleanField,
+    IntegerField,
+    FileField,
+    DirField,
+    DataField,
+    GroupField,
+)
+
+
+class ChipQC(Process):
+    """Calculate quality control metrics for ChIP-seq samples.
+
+    The analysis is based on ChIPQC package which computs a variety of
+    quality control metrics and statistics, and provides plots and
+    a report for assessment of experimental data for further analysis.
+    """
+
+    slug = 'chipqc'
+    name = 'ChipQC'
+    process_type = 'data:chipqc'
+    version = '1.0.0'
+    category = 'ChIP-Seq:QC report'
+    data_name = '{{ case|sample_name|default("?") }}'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.CACHED
+    entity = {
+        'type': 'sample',
+        'input': 'case',
+    }
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/chipseq:4.1.0_dev' # TODO
+            }
+        },
+        'resources': {
+            'cores': 8,
+            'memory': 16384,
+        },
+    }
+
+    class Input:
+        """Input fields to process ChipQC."""
+
+        alignment = DataField(
```

Comment:
It is possible that `alignment` and `peaks` would be inputs from two different samples. In that case, the above `entity` -> `input` would need to be set to one of the inputs. However, as is the case for this process, I think it is best to (if possible) check if the inputs belong to different samples, and fail the process in such cases. Does this make sense?